internal/sdkprovider/provider/provider.go

```diff
@@ -282,6 +283,12 @@ func Provider(version string) *schema.Provider {
 			return nil, diag.FromErr(err)
 		}
 
+		// fixme: temporary solution
+		err = common.CacheGenAivenClient(token, p.TerraformVersion, version)
+		if err != nil {
+			return nil, diag.FromErr(err)
+		}
```

Comment:
Why do we need `CacheGenAivenClient`? Do you already have a plan to fix this `fixme` comment?

Reply:
Due to the provider's interface limitations, I couldn't find a better way rather than create a Singleton. Eventually, it will take the place of the original client.